lnwallet/channel.go

```diff
@@ -1952,7 +1951,7 @@ func NewBreachRetribution(chanState *channeldb.OpenChannel, stateNum uint64,
 	remoteOutpoint := wire.OutPoint{
 		Hash: commitHash,
 	}
-	for i, txOut := range broadcastCommitment.TxOut {
+	for i, txOut := range revokedSnapshot.CommitTx.TxOut {
```

Comment:
It is surprising to me that we store the whole remote commit tx for each state num! Has this always been the case?

Seems a bit wasteful and not necessary?

Reply:
We considered removing them at one point and then didn't iirc. Turns out they're useful! 

> Seems a bit wasteful and not necessary?

Never hurts to keep channel data, deleting is forever. if it becomes an issue we should reconsider it, but i don't think that'll be any time soon :)

Reply:
True, must leave some low-hanging optimizations around in case we run out of work 🤣